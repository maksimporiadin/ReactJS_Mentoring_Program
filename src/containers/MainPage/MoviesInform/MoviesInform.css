.moviesInform {
    display: flex;
    justify-content: space-between;
}

.moviesInform__count {
    display: flex;
    align-items: center;
}