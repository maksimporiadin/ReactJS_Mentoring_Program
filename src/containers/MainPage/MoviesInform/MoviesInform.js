import React from 'react';

import './MoviesInform.css';

const MoviesInform = ({ count, children }) => (
    <div className="moviesInform">
        <div className="moviesInform__count">
            {(() => {
                switch (count) {
                    case 0: return null
                    case 1: return <span> {count} movie found</span>
                    default: return <span> {count} movies found</span>
                }
            })()}
        </div>
        <div className="moviesInform__sortBy">
            { children }
        </div>
    </div>
);

export default MoviesInform;
