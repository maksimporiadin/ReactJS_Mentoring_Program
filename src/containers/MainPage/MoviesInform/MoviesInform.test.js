import 'jsdom-global/register';
import React from 'react';

import MoviesInform from './index';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('MoviesInform', () => {
    let wrapper;

    it('is renders', () => {
        wrapper = shallow(<MoviesInform  />);
        expect(wrapper).toHaveLength(1);
    });
});