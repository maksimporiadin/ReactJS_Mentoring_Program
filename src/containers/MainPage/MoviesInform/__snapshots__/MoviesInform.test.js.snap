// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesInform is renders count 0 1`] = `
<div
  className="moviesInform"
>
  <div
    className="moviesInform__count"
  />
  <div
    className="moviesInform__sortBy"
  />
</div>
`;

exports[`MoviesInform is renders count 1 1`] = `
<div
  className="moviesInform"
>
  <div
    className="moviesInform__count"
  >
    <span>
       
      1
       movie found
    </span>
  </div>
  <div
    className="moviesInform__sortBy"
  />
</div>
`;

exports[`MoviesInform is renders count default 1`] = `
<div
  className="moviesInform"
>
  <div
    className="moviesInform__count"
  >
    <span>
       
      3
       movies found
    </span>
  </div>
  <div
    className="moviesInform__sortBy"
  />
</div>
`;
